sdks/java/io/bigquery-io-tests/src/test/java/org/apache/beam/sdk/bigqueryiotests/BigQueryIOIT.java

```diff
@@ -151,46 +159,51 @@ private static void createAndFillSourceTable() {
                       return tableRow;
                     })
                 .withCustomGcsTempLocation(ValueProvider.StaticValueProvider.of(tempRoot))
+                .withMethod(method)
                 .withSchema(
                     new TableSchema()
                         .setFields(
                             Collections.singletonList(
                                 new TableFieldSchema().setName("data").setType("BYTES")))));
-    pipeline.run().waitUntilFinish();
+
+    PipelineResult pipelineResult = pipeline.run();
+    pipelineResult.waitUntilFinish();
+    extractAndPublishTime(pipelineResult, WRITE_TIME_METRIC_NAME);
   }
 
-  private static boolean checkForSourceTable() {
-    BigQueryOptions options = BigQueryOptions.newBuilder().build();
-    BigQuery client = options.getService();
-    TableId tableId =
-        TableId.of(options.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
-    Table sourceTable = client.getTable(tableId, TableOption.fields(BigQuery.TableField.TYPE));
-    return sourceTable != null;
+  private void extractAndPublishTime(PipelineResult pipelineResult, String writeTimeMetricName) {
+    NamedTestResult metricResult =
+        getMetricSupplier(writeTimeMetricName).apply(new MetricsReader(pipelineResult, NAMESPACE));
+    IOITMetrics.publish(
+        TEST_ID,
+        TEST_TIMESTAMP,
+        bigQueryMetricsDataset,
+        bigQueryMetricsTable,
+        Collections.singletonList(metricResult));
   }
 
   @Test
-  public void testRead() {
+  public void testWriteThenRead() {
```

Comment:
This method should be above all private methods

Reply:
Ok, moved.